.github/scripts/jira/lib.test.ts

```diff
@@ -45,3 +64,83 @@ describe("tagsToLabels", () => {
     ]);
   });
 });
+
+const mockExecPromise = vi.fn();
+vi.mock("util", () => ({
+  promisify: () => mockExecPromise,
+}));
+
+describe("getGitTopLevel", () => {
+  it("should log the top-level directory when git command succeeds", async () => {
+    mockExecPromise.mockResolvedValueOnce({
+      stdout: "/path/to/top-level-dir",
+      stderr: "",
+    });
+
+    const mockConsoleLog = vi.spyOn(core, "info");
+    await getGitTopLevel();
+
+    expect(mockExecPromise).toHaveBeenCalledWith(
+      "git rev-parse --show-toplevel"
+    );
+    expect(mockConsoleLog).toHaveBeenCalledWith(
+      "Top-level directory: /path/to/top-level-dir"
+    );
+  });
+
+  it("should log an error message when git command fails", async () => {
+    mockExecPromise.mockRejectedValueOnce({
+      message: "Command failed",
+    });
+
+    const mockConsoleError = vi.spyOn(core, "error");
+    await getGitTopLevel().catch(() => {});
+
+    expect(mockExecPromise).toHaveBeenCalledWith(
+      "git rev-parse --show-toplevel"
+    );
+    expect(mockConsoleError).toHaveBeenCalledWith(
+      "Error executing command: Command failed"
+    );
+  });
+
+  it("should log an error message when git command output contains an error", async () => {
+    mockExecPromise.mockResolvedValueOnce({
+      stdout: "",
+      stderr: "Error: Command failed",
+    });
+
+    const mockConsoleError = vi.spyOn(core, "error");
+    await getGitTopLevel().catch(() => {});
+
+    expect(mockExecPromise).toHaveBeenCalledWith(
+      "git rev-parse --show-toplevel"
+    );
+    expect(mockConsoleError).toHaveBeenCalledWith(
+      "Error in command output: Error: Command failed"
+    );
+  });
+});
+
+describe("generateJiraIssuesLink", () => {
+  it("should generate a Jira issues link", () => {
+    expect(
+      generateJiraIssuesLink(
+        "review-artifacts-automation-base:0de9b3b-head:e5b3b9d"
+      )
+    ).toMatchInlineSnapshot(
+      `"https://smartcontract-it.atlassian.net/issues/?jql=labels+%3D+%22review-artifacts-automation-base%3A0de9b3b-head%3Ae5b3b9d%22"`
+    );
+  });
+});
+
+describe("generateIssueLabel", () => {
+  it("should generate an issue label", () => {
+    const product = "automation";
+    const baseRef = "0de9b3b";
+    const headRef = "e5b3b9d";
+    expect(generateIssueLabel(product, baseRef, headRef)).toMatchInlineSnapshot(
+      `"review-artifacts-automation-base:0de9b3b-head:e5b3b9d"`
+    );
+  });
+});
```

Comment:
would it make sense to test things like:
* no changeset found
* too many changesets found
* various JIRA errors (comment already exists, one of the issues is missing, etc?)

Reply:
Yeah, these have already been tested manually.